guava/src/com/google/common/escape/CharEscaper.java

```diff
@@ -93,14 +97,22 @@ public String escape(String string) {
    * @return the escaped form of {@code string}
    * @throws NullPointerException if {@code string} is null
    */
-  protected final String escapeSlow(String s, int index) {
+  @SuppressWarnings(value = {"upperbound:assignment.type.incompatible", "upperbound:argument.type.incompatible",/*
+  line 132 and line 152: Because of System.arraycopy() method, `rlen` is required to be
+   @LTLengthOf(value={"r", "dest"}, offset={"-1", "destIndex - 1"}).
+   Since rlen = r.length, rlen should already be inferred to have length of @LTLengthOf(value="r", offset="-1" */
```

Comment:
Have you tried actually writing this annotation? We discussed last week how the typechecker throws away extra information if there's a user-written type on a local variable, so you should try writing that type and see if the error goes away (which it should).